src/Stack/Package.hs

```diff
@@ -732,7 +735,7 @@ resolveGlobFiles =
         mapM resolveFileOrWarn names
     matchDirFileGlob' dir glob =
         catch
-            (matchDirFileGlob_ dir glob)
+            (liftIO (matchDirFileGlob minBound cabalFileVersion dir glob))
```

Comment:
Stack previously used a patched version of this function from Cabal, removed in this same diff, which allowed the extensions to be a suffix rather than matching strictly exactly. This behaviour is now present in Cabal's functions, but *only* if the spec version passed in is at least 2.4, so any older packages may see a behaviour change here.

On the other hand, this is now consistent with cabal-install's behaviour and what you get from `runghc Setup.hs`, and also corresponds to what the Cabal docs say and have said for ages, so I think this ought to be OK. Even so, I'd mention it in the changelog.